projects/docker-cloud/e2e/init_test.go

```diff
@@ -0,0 +1,60 @@
+package e2e_test
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+)
+
+func TestMain(m *testing.M) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not construct pool: %s", err)
+	}
+
+	// uses pool to try to connect to Docker
+	err = pool.Client.Ping()
+
+	if err != nil {
+		log.Fatalf("Could not connect to Docker: %s", err)
+	}
+
+	// Build and run the given Dockerfile
+	resource, err := pool.BuildAndRunWithBuildOptions(&dockertest.BuildOptions{
+		ContextDir: "../",
+		Dockerfile: "Dockerfile",
+	}, &dockertest.RunOptions{
+		Name:         "docker-cloud-e2e",
+		ExposedPorts: []string{"80/tcp"},
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			"80/tcp": {{HostPort: "80"}},
+		}})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	if err = pool.Retry(func() error {
+		fmt.Println("Checking API connection...")
+		_, err := http.Get("http://localhost:80/")
```

Comment:
Rather than repeating 80 both here and on line 35, I'd consider extracting a variable for the host port number to show they're meant to be the same (and if one changes, so should the other)

Note: I wouldn't extract _both_ of the 80s on line 35 into the variable - the port _inside the container_ is _not_ specifically tied to the port _exposed to the host_ - extracting all three to the same variable is probably not correct.

Line 29 of projects/docker-cloud/e2e/main_test.go should probably _also_ share that variable (which means maybe it should exist as an environment variable or some kind of global mutable variable that both TestMain and your tests can access)